src/openapi.json

```diff
@@ -0,0 +1,566 @@
+        "parameters": [
+          { "$ref": "#/components/parameters/canonical_field_label"}
+        ],
+        "responses": {
+          "200": {
+            "description": "Canonical field.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object"
+                }
+              }
+            }
+          }
+        }
+      },
+      "put": {
+        "description": "Adds a canonical field idempotently or returns a 409 error if change to an existing canonical field was attempted.",
+        "parameters": [
+          { "$ref": "#/components/parameters/canonical_field_label"}
+        ],
+        "requestBody": {
+          "required": true,
+          "content": {
+            "application/json:": {
+              "schema": {
+                "type": "object"
+              }
+            }
+          }
+        },
+        "responses": {
+          "201": {
+            "description": "The (new) canonical field that was just created.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object"
+                }
+              }
+            }
+          },
+          "200": {
+            "description": "The (identical, existing) canonical field.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object"
+                }
+              }
+            }
+          },
+          "409": {
+            "description": "The canonical field label already existed but with different attributes. Existing canonical fields cannot be replaced with different attributes, they are immutable after first creation.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object"
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/opportunities": {
+      "summary": "Funding opportunities such as an available grant.",
+      "get": {
+        "responses": {
+          "200": {
+            "description": "A list of known funding opportunities.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array"
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/opportunities/{opportunity_title}": {
+      "summary": "A funding opportunity such as an available grant, RFP, etc.",
+      "get": {
+        "parameters": [
+          { "$ref": "#/components/parameters/opportunity_title"}
+        ],
+        "responses": {
+          "200": {
+            "description": "The funding opportunity.",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object"
+                }
+              }
+            }
+          }
+        }
+      },
+      "put": {
```

Comment:
Same as canonical fields -- `POST` is the REST verb for create (and we don't want `PUT` to be implemented since we aren't allowing updates -- so it would 404 appropriately.)